src/OSPSuite.Presentation/Mappers/DiffItemToDiffItemDTOMapper.cs

```diff
@@ -86,7 +89,24 @@ private string presentWithDetails(string presentDetails)
 
       public void Visit(PropertyValueDiffItem propertyDiffItem)
       {
-         updateDiffItem(propertyDiffItem.FormattedValue1, propertyDiffItem.FormattedValue2, propertyDiffItem.PropertyName, objectNameFrom(propertyDiffItem), propertyDiffItem.CommonAncestor, itemIsMissing: false);
+         Func<PathElements> pathElementsMapper = () => mapPathElementsAsIEntity(propertyDiffItem.CommonAncestor);
+
+         if (propertyDiffItem.Object1.GetType() == propertyDiffItem.Object2.GetType() && propertyDiffItem.Object1.IsAnImplementationOf<PathAndValueEntity>())
+            pathElementsMapper = () => mapPathElementsAsPathAndValueEntity(propertyDiffItem);
+
+         updateDiffItem(propertyDiffItem.FormattedValue1, propertyDiffItem.FormattedValue2, propertyDiffItem.PropertyName, objectNameFrom(propertyDiffItem), itemIsMissing: false, pathElementsMapper);
+      }
+
+      private PathElements mapPathElementsAsPathAndValueEntity(PropertyValueDiffItem propertyDiffItem)
+      {
+         var object1 = propertyDiffItem.Object1 as PathAndValueEntity;
+         var object2 = propertyDiffItem.Object2 as PathAndValueEntity;
+
+         if (object1 == null || object2 == null || !string.Equals(object2.Path.PathAsString, object1.Path.PathAsString))
```

Comment:
And the second condition? it this check required? Isn't it how we create those objects in the first place. If teels like line 95 could be replaced with ()=> _pathAndValueEntityToPathElementsMapper.MapFrom(object1); and this method gone.

Reply:
I wanted to avoid the implied knowledge of how PathAndValueEntity equivalency is determined, but we do that with Name in other DiffItems all the time, so this is the same, but in this case, with Path.